Source/Add-ins/Profile.Common/ModalWindows/ObservationPointsListModalWindow.cs

```diff
@@ -1,47 +1,114 @@
-﻿using ESRI.ArcGIS.Geometry;
+﻿using ESRI.ArcGIS.Carto;
+using ESRI.ArcGIS.Geodatabase;
+using ESRI.ArcGIS.Geometry;
+using MilSpace.Core;
 using MilSpace.Core.Tools;
-using MilSpace.DataAccess.DataTransfer;
+using MilSpace.Profile.Localization;
 using System;
 using System.Collections.Generic;
-using System.ComponentModel;
-using System.Data;
-using System.Drawing;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Runtime.InteropServices;
 using System.Windows.Forms;
 
 namespace MilSpace.Profile.ModalWindows
 {
     public partial class ObservationPointsListModalWindow : Form
     {
-        private List<ObservationPoint> _points = new List<ObservationPoint>();
+        private List<FromLayerPointModel> _points = new List<FromLayerPointModel>();
+        private Logger _log = Logger.GetLoggerEx("MilSpace.Profile.ModalWindows.ObservationPointsListModalWindow");
         public IPoint SelectedPoint; 
 
-        public ObservationPointsListModalWindow(IEnumerable<ObservationPoint> points)
+        public ObservationPointsListModalWindow(IFeatureLayer observPointsLayer)
         {
             InitializeComponent();
+            LocalizeStrings();
+            FillPointsGrid(observPointsLayer);
+        }
 
-            _points = points.ToList();
-            FillPointsGrid(points);
+        private void LocalizeStrings()
+        {
+            this.Text = LocalizationContext.Instance.FindLocalizedElement("ModalObservPointsTitle", "Вибір точки з шару точок спостереження");
+            btnChoosePoint.Text = LocalizationContext.Instance.FindLocalizedElement("BtnChooseText", "Обрати");
+            dgvPoints.Columns["IdCol"].HeaderText = LocalizationContext.Instance.FindLocalizedElement("DgvObservPointsIdHeader", "Ідентифікатор");
+            dgvPoints.Columns["TitleCol"].HeaderText = LocalizationContext.Instance.FindLocalizedElement("DgvObservPointsTitleHeader", "Назва");
         }
 
-        private void FillPointsGrid(IEnumerable<ObservationPoint> points)
+        private void FillPointsGrid(IFeatureLayer layer)
         {
-            dgvPoints.Rows.Clear();
+            var featureClass = layer.FeatureClass;
```

Comment:
Form code should not contains any data reading code. Just fields\lists filling. This code should be moved to the  interaction module where a method  will return just a list of objects